osgp/protocol-adapter-oslp/oslp/src/main/java/org/opensmartgridplatform/oslp/OslpUtils.java

```diff
@@ -165,23 +166,51 @@ public static boolean validateSignature(final byte[] message, final byte[] secur
         final Signature signatureBuilder = Signature.getInstance(signature, provider);
         signatureBuilder.initVerify(publicKey);
         signatureBuilder.update(message);
-        
+
         int signatureLength = securityKey.length;
-        
+
         if (ArrayUtils.contains(TRUNCATE_SIGNATURES, signature)) {
             // Fix for https://bugs.openjdk.java.net/browse/JDK-8161571
-            // Read 2nd byte as length indicator for the actual signature bytes, include 2 bytes for 1st 2 bytes
-            // Ensure the byte (which is signed) is converted correctly to a positive int
-            signatureLength = securityKey[1]+2 & 0xFF;
+            // Read 2nd byte as length indicator for the actual signature bytes,
+            // include 2 bytes for 1st 2 bytes
+            // Ensure the byte (which is signed) is converted correctly to a
+            // positive int
+            signatureLength = securityKey[1] + 2 & 0xFF;
             if (signatureLength > securityKey.length) {
-                throw new GeneralSecurityException("Size indicator in ASN.1 DSA signature to large [" + signatureLength + "]");
+                throw new GeneralSecurityException(
+                        "Size indicator in ASN.1 DSA signature to large [" + signatureLength + "]");
             }
         }
         // Truncate the string to actual ASN.1 DSA length, removing padding
-        byte[] truncated = Arrays.copyOf(securityKey, signatureLength);
+        final byte[] truncated = Arrays.copyOf(securityKey, signatureLength);
         return signatureBuilder.verify(truncated);
     }
 
+    public static boolean isOslpResponse(final OslpEnvelope envelope) {
+        return envelope.getPayloadMessage().hasRegisterDeviceResponse()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (21) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1067)